src/acl/DomainData.cc

```diff
@@ -79,22 +79,22 @@ aclDomainCompare(T const &a, T const &b)
             // sub-domain is ignored. That is okay. Just important
             bool d3big = (strlen(d3) > strlen(d4)); // Always suggest removing the longer one.
             debugs(28, DBG_IMPORTANT, "WARNING: '" << (d3big?d3:d4) << "' is a subdomain of '" << (d3big?d4:d3) << "'");
-            debugs(28, DBG_IMPORTANT, "WARNING: You should remove '" << (d3big?d3:d4) << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, DBG_IMPORTANT, "WARNING: You should remove '" << (d3big?d3:d4) << "' from the ACL");
             debugs(28, 2, "Ignore '" << d3 << "' to keep splay tree searching predictable");
         }
     } else if (ret == 0) {
         // It may be an exact duplicate. No problem. Just drop.
         if (strcmp(d1,d2)==0) {
             debugs(28, 2, "WARNING: '" << d2 << "' is duplicated in the list.");
-            debugs(28, 2, "WARNING: You should remove one '" << d2 << "' from the ACL named '" << AclMatchedName << "'");
+            debugs(28, 2, "WARNING: You should remove one '" << d2 << "' from the ACL");
             return ret;
         }
         // When a.example.com comes before .example.com in an ACL
         // discarding the wildcard is critically bad.
         // or Maybe even both are wildcards. Things are very weird in those cases.
         bool d1big = (strlen(d1) > strlen(d2)); // Always suggest removing the longer one.
         debugs(28, DBG_CRITICAL, "ERROR: '" << (d1big?d1:d2) << "' is a subdomain of '" << (d1big?d2:d1) << "'");
-        debugs(28, DBG_CRITICAL, "ERROR: You need to remove '" << (d1big?d1:d2) << "' from the ACL named '" << AclMatchedName << "'");
+        debugs(28, DBG_CRITICAL, "ERROR: You need to remove '" << (d1big?d1:d2) << "' from the ACL");
```

Comment:
With our CodeContext changes, Squid started to create noise by printing identical context information several times (for each debugs() line and self_destruct() below). We could reduce this noise by joining these two lines (at least).

Reply:
Yes, please adjust/merge these debugs() into one in this PR. Same for nearby debugs().

Reply:
Done (7e0237b), though I did not change the message phrasing much.

Reply:
However, this adjustment only slightly reduced the noise - leaving out 5 other duplicated context entries. I guess we need to throw instead self_destruct() here and CallService() instead of CallContextCreator() in Acl.cc. With these changes, CodeContext will be reset back to nil automatically before the caught exception leads to another self_destruct().